code-challenges/challenges-04/challenges-04.test.js

```diff
@@ -0,0 +1,224 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named isNum that takes in a string or number of any length. This function should use a regular expression pattern to return true if the input contains a number, and false if the input does not contain a number.
+
+For example:
+12345 returns true
+'12345' returns true
+'h3llo world' returns true
+'hello world' returns false
+------------------------------------------------------------------------------------------------ */
+
+const isNum = (input) => {
+    let isNumber = /[0-9]/g;
+    return isNumber.test(input);
+  // Solution code here...
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named isCapitalized that takes in a string. This function should use a regular expression pattern to match all words that begin with a capital letter. It should only match words, not punctuation.
+
+Return an array containing all the matches.
+------------------------------------------------------------------------------------------------ */
+
+const isCapitalized = (str) => {
+
+    const regex = /\b[A-Z]\w*/g;
+  const matches = str.match(regex)
+  return matches ? matches :[]; ///// by internt search //////
+
+    // Solution code here...
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named citiesAtoJ that takes in an array of city names and uses a regular expression pattern to return a new array containing any cities that begin with the letters A through J, inclusive.
+------------------------------------------------------------------------------------------------ */
+
+const citiesAtoJ = (arr) => {
+  // Solution code here...
+  const cityAJ = /^[A-J]\w*/g;
+  const cityArray = [];
+  arr.forEach(element => {
+  if (element.match(cityAJ))
+   cityArray.push(element);
+  });
+  return cityArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+You have created a game application and begin by asking users an easy question: In which month is Halloween?
+
+Write a function named matchMonth which uses a regular expression pattern to match any of these inputs: October, Oct, october, oct
+
+If the user enters any of these four inputs, return true. For any other input, return false.
+
+Do not use the vertical bar (pipe) in your pattern.
+------------------------------------------------------------------------------------------------ */
+
+const matchMonth = (input) => {
+    const monthm = /^[Oo]ct(ober)?$/; //// by someone else code //////
```

Comment:
I need you to cite your source. "someone else's code" doesn't tell me who. I need a link to the page where you found it.

And, I hope you can begin to see how these expresssions work. They are complicated, but they can really help